### PR TITLE
AB#74543 added land status page

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

File: src/HE.Investment.AHP.Contract/FinancialDetails/ValueObjects/PurchasePrice.cs [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
+            OperationResult.New()
+                .AddValidationError(FinancialDetailsValidationFieldNames.PurchasePrice, FinancialDetailsValidationErrors.InvalidPurchasePrice)
+                .CheckErrors();
+        }
+
         Value = value;
-        IsFinal = isFinal;
     }
 
     public string Value { get; }
```
Comment (Carq) [inline src/HE.Investment.AHP.Contract/FinancialDetails/ValueObjects/PurchasePrice.cs:29]: Value should be int

File: src/HE.Investment.AHP.Domain/FinancialDetails/Entities/FinancialDetailsEntity.cs [hunk 2]
```diff
@@ -4,18 +4,21 @@ namespace HE.Investment.AHP.Domain.FinancialDetails.Entities;
...
     }
 
     public FinancialSchemeId FinancialSchemeId { get; }
 
-    public FinancialDetailsId Id { get; }
+    public FinancialDetailsId FinancialDetailsId { get; }
 
     public PurchasePrice PurchasePrice { get; set; }
 
+    public bool IsPurchasePriceKnown { get; set; }
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/FinancialDetails/Entities/FinancialDetailsEntity.cs:20]: All properties should have private setters

File: src/HE.Investment.AHP.Contract/FinancialDetails/ValueObjects/PurchasePrice.cs (hunk 1 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Contract/FinancialDetails/ValueObjects/PurchasePrice.cs]: I've chnages type as suggested

File: src/HE.Investment.AHP.Domain/FinancialDetails/Entities/FinancialDetailsEntity.cs (hunk 2 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/FinancialDetails/Entities/FinancialDetailsEntity.cs]: Done

**Review by Antek012 — APPROVED** [review]

**Review by jjankowski87 — APPROVED** [review]

**Pull request merged**